llama-index-integrations/vector_stores/llama-index-vector-stores-postgres/llama_index/vector_stores/postgres/base.py

```diff
@@ -380,16 +380,19 @@ def _to_postgres_operator(self, operator: FilterOperator) -> str:
             return "="
 
     def _build_filter_clause(self, filter_: MetadataFilter) -> Any:
-        from sqlalchemy import text, bindparam
+        from sqlalchemy import text
 
         if filter_.operator in [FilterOperator.IN, FilterOperator.NIN]:
             # Expects a single value in the metadata, and a list to compare
-            return (
-                text(
-                    f"metadata_->>'{filter_.key}' {self._to_postgres_operator(filter_.operator)} :values"
-                )
-                .bindparams(bindparam("values", expanding=True))
-                .bindparams(values=tuple(filter_.value))
+
+            # In Python, to create a tuple with a single element, you need to include a comma after the element
+            # This code will correctly format the IN clause whether there is one element or multiple elements in the list:
+            filter_value = ", ".join(f"'{e}'" for e in filter_.value)
+
+            return text(
+                f"metadata_->>'{filter_.key}' "
+                f"{self._to_postgres_operator(filter_.operator)} "
+                f"({filter_value})"
```

Comment:
any reason to remove the bindparams here? Technically doesn't this allow SQL injection now?

Reply:
well, maybe not, but I'd feel safer with bindparams if possible 👀

Reply:
Well, if you see the rest of the code, only the IN operator that used to use bind params, the rest of them are not.

```
        elif filter_.operator == FilterOperator.CONTAINS:
            # Expects a list stored in the metadata, and a single value to compare
            return text(
                f"metadata_::jsonb->'{filter_.key}' "
                f"{self._to_postgres_operator(filter_.operator)} "
                f"'[\"{filter_.value}\"]'"
            )
        else:
            # Check if value is a number. If so, cast the metadata value to a float
            # This is necessary because the metadata is stored as a string
            try:
                return text(
                    f"(metadata_->>'{filter_.key}')::float "
                    f"{self._to_postgres_operator(filter_.operator)} "
                    f"{float(filter_.value)}"
                )
            except ValueError:
                # If not a number, then treat it as a string
                return text(
                    f"metadata_->>'{filter_.key}' "
                    f"{self._to_postgres_operator(filter_.operator)} "
                    f"'{filter_.value}'"
                )

```

Reply:
eh, fair